### ISSUE TITLE
[🐛 Bug]: import type {Options} from '@wdio/types' ^^^^^^  SyntaxError: Cannot use import statement outside a module

### ISSUE THREAD

**Issue by SKannoth** [webdriverio/webdriverio#10944]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

wdio/cli": "^8.14.6",

### Node.js Version

v20.3.1

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
package.json
{
  "name": "webdriverio",
  "version": "1.0.0",
  "description": "test",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "wdio": "wdio run ./wdio.conf.ts"
  },
  "author": "Shijith",
  "license": "ISC",
  "dependencies": {
    "@wdio/types": "^8.14.0",
    "wdio": "^6.0.1"
  },
  "devDependencies": {
    "@wdio/allure-reporter": "^8.14.6",
    "@wdio/cli": "^8.14.6",
    "@wdio/cucumber-framework": "^8.14.6",
    "@wdio/local-runner": "^8.14.6",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6",
    "wdio-cucumberjs-json-reporter": "^5.1.6"
  }
}

wdio.conf.ts:
import type {Options} from '@wdio/types'
export const config: Options.Testrunner = {
    //
    // ==================== 
    // Runner Configuration
    // ====================
    // WebdriverIO supports running e2e tests as well as unit and component tests.
    runner: 'local',
    autoCompileOpts: {
        autoCompile: true,
        tsNodeOpts: {
            project: './tsconfig.json',
            transpileOnly: true
        }
    },
    
    //
    // ==================
    // Specify Test Files
    // ==================
    // Define which test specs should run. The pattern is relative to the directory
    // of the configuration file being run.
    //
    // The specs are defined as an array of spec files (optionally using wildcards
    // that will be expanded). The test for each spec file will be run in a separate
    // worker process. In order to have a group of spec files run in the same worker
    // process simply enclose them in an array within the specs array.
    //
    // If you are calling `wdio` from an NPM script (see https://docs.npmjs.com/cli/run-script),
    // then the current working directory is where your `package.json` resides, so `wdio`
    // will be called from there.
    //
    specs: [
        './features/**/*.feature'
    ],
    // Patterns to exclude.
    exclude: [
        // 'path/to/excluded/files'
    ],
    //
    // ============
    // Capabilities
    // ============
    // Define your capabilities here. WebdriverIO can run multiple capabilities at the same
    // time. Depending on the number of capabilities, WebdriverIO launches several test
    // sessions. Within your capabilities you can overwrite the spec and exclude options in
    // order to group specific specs to a specific capability.
    //
    // First, you can define how many instances should be started at the same time. Let's
    // say you have 3 different capabilities (Chrome, Firefox, and Safari) and you have
    // set maxInstances to 1; wdio will spawn 3 processes. Therefore, if you have 10 spec
    // files and you set maxInstances to 10, all spec files will get tested at the same time
    // and 30 processes will get spawned. The property handles how many capabilities
    // from the same test should run tests.
    //
    maxInstances: 10,
    //
    // If you have trouble getting all important capabilities together, check out the
    // Sauce Labs platform configurator - a great tool to configure your capabilities:
    // https://saucelabs.com/platform/platform-configurator
    //
    capabilities: [{
        browserName: 'chrome'
    }],

    //
    // ===================
    // Test Configurations
    // ===================
    // Define all options that are relevant for the WebdriverIO instance here
    //
    // Level of logging verbosity: trace | debug | info | warn | error | silent
    logLevel: 'info',
    //
    // Set specific log levels per logger
    // loggers:
    // - webdriver, webdriverio
    // - @wdio/browserstack-service, @wdio/devtools-service, @wdio/sauce-service
    // - @wdio/mocha-framework, @wdio/jasmine-framework
    // - @wdio/local-runner
    // - @wdio/sumologic-reporter
    // - @wdio/cli, @wdio/config, @wdio/utils
    // Level of logging verbosity: trace | debug | info | warn | error | silent
    // logLevels: {
    //     webdriver: 'info',
    //     '@wdio/appium-service': 'info'
    // },
    //
    // If you only want to run your tests until a specific amount of tests have failed use
    // bail (default is 0 - don't bail, run all tests).
    bail: 0,
    //
    // Set a base URL in order to shorten url command calls. If your `url` parameter starts
    // with `/`, the base url gets prepended, not including the path portion of your baseUrl.
    // If your `url` parameter starts without a scheme or `/` (like `some/path`), the base url
    // gets prepended directly.
    baseUrl: 'http://localhost',
    //
    // Default timeout for all waitFor* commands.
    waitforTimeout: 10000,
    //
    // Default timeout in milliseconds for request
    // if browser driver or grid doesn't send response
    connectionRetryTimeout: 120000,
    //
    // Default request retries count
    connectionRetryCount: 3,
    //
    // Test runner services
    // Services take over a specific job you don't want to take care of. They enhance
    // your test setup with almost no effort. Unlike plugins, they don't add new
    // commands. Instead, they hook themselves up into the test process.
    // services: [],
    //
    // Framework you want to run your specs with.
    // The following are supported: Mocha, Jasmine, and Cucumber
    // see also: https://webdriver.io/docs/frameworks
    //
    // Make sure you have the wdio adapter package for the specific framework installed
    // before running any tests.
    framework: 'cucumber',
    //
    // The number of times to retry the entire specfile when it fails as a whole
    // specFileRetries: 1,
    //
    // Delay in seconds between the spec file retry attempts
    // specFileRetriesDelay: 0,
    //
    // Whether or not retried spec files should be retried immediately or deferred to the end of the queue
    // specFileRetriesDeferred: false,
    //
    // Test reporter for stdout.
    // The only one supported by default is 'dot'
    // see also: https://webdriver.io/docs/dot-reporter
    reporters: [['allure', {outputDir: 'allure-results'}],'cucumberjs-json'],

    //
    // If you are using Cucumber you need to specify the location of your step definitions.
    cucumberOpts: {
        // <string[]> (file/dir) require files before executing features
        require: ['./features/step-definitions/steps.ts'],
        // <boolean> show full backtrace for errors
        backtrace: false,
        // <string[]> ("extension:module") require files with the given EXTENSION after requiring MODULE (repeatable)
        requireModule: [],
        // <boolean> invoke formatters without executing steps
        dryRun: false,
        // <boolean> abort the run on first failure
        failFast: false,
        // <boolean> hide step definition snippets for pending steps
        snippets: true,
        // <boolean> hide source uris
        source: true,
        // <boolean> fail if there are any undefined or pending steps
        strict: false,
        // <string> (expression) only execute the features or scenarios with tags matching the expression
        tagExpression: '',
        // <number> timeout for step definitions
        timeout: 60000,
        // <boolean> Enable this config to treat undefined definitions as warnings.
        ignoreUndefinedDefinitions: false
    },
    
    //
    // =====
    // Hooks
    // =====
    // WebdriverIO provides several hooks you can use to interfere with the test process in order to enhance
    // it and to build services around it. You can either apply a single function or an array of
    // methods to it. If one of them returns with a promise, WebdriverIO will wait until that promise got
    // resolved to continue.
    /**
     * Gets executed once before all workers get launched.
     * @param {object} config wdio configuration object
     * @param {Array.<Object>} capabilities list of capabilities details
     */
    // onPrepare: function (config, capabilities) {
    // },
    /**
     * Gets executed before a worker process is spawned and can be used to initialise specific service
     * for that worker as well as modify runtime environments in an async fashion.
     * @param  {string} cid      capability id (e.g 0-0)
     * @param  {object} caps     object containing capabilities for session that will be spawn in the worker
     * @param  {object} specs    specs to be run in the worker process
     * @param  {object} args     object that will be merged with the main configuration once worker is initialized
     * @param  {object} execArgv list of string arguments passed to the worker process
     */
    // onWorkerStart: function (cid, caps, specs, args, execArgv) {
    // },
    /**
     * Gets executed just after a worker process has exited.
     * @param  {string} cid      capability id (e.g 0-0)
     * @param  {number} exitCode 0 - success, 1 - fail
     * @param  {object} specs    specs to be run in the worker process
     * @param  {number} retries  number of retries used
     */
    // onWorkerEnd: function (cid, exitCode, specs, retries) {
    // },
    /**
     * Gets executed just before initialising the webdriver session and test framework. It allows you
     * to manipulate configurations depending on the capability or spec.
     * @param {object} config wdio configuration object
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {Array.<String>} specs List of spec file paths that are to be run
     * @param {string} cid worker id (e.g. 0-0)
     */
    // beforeSession: function (config, capabilities, specs, cid) {
    // },
    /**
     * Gets executed before test execution begins. At this point you can access to all global
     * variables like `browser`. It is the perfect place to define custom commands.
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {Array.<String>} specs        List of spec file paths that are to be run
     * @param {object}         browser      instance of created browser/device session
     */
    // before: function (capabilities, specs) {
    // },
    /**
     * Runs before a WebdriverIO command gets executed.
     * @param {string} commandName hook command name
     * @param {Array} args arguments that command would receive
     */
    // beforeCommand: function (commandName, args) {
    // },
    /**
     * Cucumber Hooks
     *
     * Runs before a Cucumber Feature.
     * @param {string}                   uri      path to feature file
     * @param {GherkinDocument.IFeature} feature  Cucumber feature object
     */
    // beforeFeature: function (uri, feature) {
    // },
    /**
     *
     * Runs before a Cucumber Scenario.
     * @param {ITestCaseHookParameter} world    world object containing information on pickle and test step
     * @param {object}                 context  Cucumber World object
     */
    // beforeScenario: function (world, context) {
    // },
    /**
     *
     * Runs before a Cucumber Step.
     * @param {Pickle.IPickleStep} step     step data
     * @param {IPickle}            scenario scenario pickle
     * @param {object}             context  Cucumber World object
     */
    // beforeStep: function (step, scenario, context) {
    // },
    /**
     *
     * Runs after a Cucumber Step.
     * @param {Pickle.IPickleStep} step             step data
     * @param {IPickle}            scenario         scenario pickle
     * @param {object}             result           results object containing scenario results
     * @param {boolean}            result.passed    true if scenario has passed
     * @param {string}             result.error     error stack if scenario failed
     * @param {number}             result.duration  duration of scenario in milliseconds
     * @param {object}             context          Cucumber World object
     */
    // afterStep: function (step, scenario, result, context) {
    // },
    /**
     *
     * Runs after a Cucumber Scenario.
     * @param {ITestCaseHookParameter} world            world object containing information on pickle and test step
     * @param {object}                 result           results object containing scenario results
     * @param {boolean}                result.passed    true if scenario has passed
     * @param {string}                 result.error     error stack if scenario failed
     * @param {number}                 result.duration  duration of scenario in milliseconds
     * @param {object}                 context          Cucumber World object
     */
    // afterScenario: function (world, result, context) {
    // },
    /**
     *
     * Runs after a Cucumber Feature.
     * @param {string}                   uri      path to feature file
     * @param {GherkinDocument.IFeature} feature  Cucumber feature object
     */
    // afterFeature: function (uri, feature) {
    // },
    
    /**
     * Runs after a WebdriverIO command gets executed
     * @param {string} commandName hook command name
     * @param {Array} args arguments that command would receive
     * @param {number} result 0 - command success, 1 - command error
     * @param {object} error error object if any
     */
    // afterCommand: function (commandName, args, result, error) {
    // },
    /**
     * Gets executed after all tests are done. You still have access to all global variables from
     * the test.
     * @param {number} result 0 - test pass, 1 - test fail
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {Array.<String>} specs List of spec file paths that ran
     */
    // after: function (result, capabilities, specs) {
    // },
    /**
     * Gets executed right after terminating the webdriver session.
     * @param {object} config wdio configuration object
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {Array.<String>} specs List of spec file paths that ran
     */
    // afterSession: function (config, capabilities, specs) {
    // },
    /**
     * Gets executed after all workers got shut down and the process is about to exit. An error
     * thrown in the onComplete hook will result in the test run failing.
     * @param {object} exitCode 0 - success, 1 - fail
     * @param {object} config wdio configuration object
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {<Object>} results object containing test results
     */
    // onComplete: function(exitCode, config, capabilities, results) {
    // },
    /**
    * Gets executed when a refresh happens.
    * @param {string} oldSessionId session ID of the old session
    * @param {string} newSessionId session ID of the new session
    */
    // onReload: function(oldSessionId, newSessionId) {
    // }
}

tsconfig.json
{
    "compilerOptions": {
        "moduleResolution": "node",
        "module": "commonjs",
        "target": "es2022",
        "types": [
            "node",
            "@wdio/globals/types",
            "expect-webdriverio",
            "@wdio/cucumber-framework"
        ],
        "skipLibCheck": true,
        "noEmit": true,
        "allowImportingTsExtensions": true,
        "resolveJsonModule": true,
        "isolatedModules": true,
        "strict": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noFallthroughCasesInSwitch": true
    },
    "include": [
        "src"
    ]
}
```

### What happened?

When I try to run the script it gives me the below error 

 npx wdio wdio.conf.ts
2023-08-14T15:23:56.014Z ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///****/webdriverio/wdio.conf.ts: Cannot use import statement outside a module
***webdriverio/wdio.conf.ts:1
import type {Options} from '@wdio/types'
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1177:20)
    at Module._compile (node:internal/modules/cjs/loader:1221:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1311:10)
    at Module.load (node:internal/modules/cjs/loader:1115:32)
    at Module._load (node:internal/modules/cjs/loader:962:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:165:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:192:25)
    at async CustomizedModuleLoader.import (node:internal/modules/esm/loader:246:24)
    at async ConfigParser.addConfigFile (file:///******/webdriverio/node_modules/@wdio/config/build/lib/ConfigParser.js:85:36)
    at async ConfigParser.initialize (file://*****/webdriverio/node_modules/@wdio/config/build/lib/ConfigParser.js:47:13)
    at async Launcher.run (file:***/webdriverio/node_modules/@wdio/cli/build/launcher.js:37:9)
    at async Module.run (file:****webdriverio/node_modules/@wdio/cli/build/run.js:64:20)

### What is your expected behavior?

expecting the script to run and execute the chrome browser

### How to reproduce the bug.

npm init  - 
npm install wdio 
npx wdio wdio.conf.ts

### Relevant log output

```typescript
When I try to run the script it gives me the below error 

 npx wdio wdio.conf.ts
2023-08-14T15:23:56.014Z ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///****/webdriverio/wdio.conf.ts: Cannot use import statement outside a module
***webdriverio/wdio.conf.ts:1
import type {Options} from '@wdio/types'
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1177:20)
    at Module._compile (node:internal/modules/cjs/loader:1221:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1311:10)
    at Module.load (node:internal/modules/cjs/loader:1115:32)
    at Module._load (node:internal/modules/cjs/loader:962:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:165:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:192:25)
    at async CustomizedModuleLoader.import (node:internal/modules/esm/loader:246:24)
    at async ConfigParser.addConfigFile (file:///******/webdriverio/node_modules/@wdio/config/build/lib/ConfigParser.js:85:36)
    at async ConfigParser.initialize (file://*****/webdriverio/node_modules/@wdio/config/build/lib/ConfigParser.js:47:13)
    at async Launcher.run (file:***/webdriverio/node_modules/@wdio/cli/build/launcher.js:37:9)
    at async Module.run (file:****webdriverio/node_modules/@wdio/cli/build/run.js:64:20)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!

I have been running against this issue myself and it is caused by the way we integrated `ts-node`. Until this is resolved I recommend as a workaround to make the project an ESM project by adding `"type": "module"` to the `package.json` and updated `"module": "ESNext",` in `tsconfig.json`.

We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by christian-bromann** [comment]
@jan-molak just fyi: this is the edge case that required me to set `-r ts-node/register` to the `NODE_FLAGS` environment. Do you happen to know any easy way to detect whether a project is running as ESM or CJS project? Because based on that we should set one or the other ts-node hook. There doesn't seem to be one for both.

**Comment by SKannoth** [comment]
Thanks @christian-bromann

**Comment by jan-molak** [comment]
> @jan-molak just fyi: this is the edge case that required me to set `-r ts-node/register` to the `NODE_FLAGS` environment. Do you happen to know any easy way to detect whether a project is running as ESM or CJS project? Because based on that we should set one or the other ts-node hook. There doesn't seem to be one for both.

Yeah, I think it's either CJS or ESM, there's no setting that I know of that supports both.

I think the only way to reliably detect it is to parse the consumer's package.json and look for type: module. If it's there, we have an ESM project, if not - CJS. Do you think this could work, @christian-bromann ?

**Comment by christian-bromann** [comment]
Maybe we can use something like https://github.com/unjs/mlly to detect it.

**Comment by christian-bromann** [comment]
@SKannoth can you provide a reproducible example? I am having a hard time reproducing this.

**Comment by SKannoth** [comment]
@christian-bromann I just used the standard instruction given for typescript setup with cucumber.
npm init
npm install @wdio/cli
npx wdio config
Output:
===============================
🤖 WDIO Configuration Wizard 🧙
===============================

? A project named "issue-reproduce" was detected at "/Users/shijithkannoth/Projects/POC/issue-reproduce",
correct? Yes
? What type of testing would you like to do? E2E Testing - of Web or Mobile Applications
? Where is your automation backend located? On my local machine
? Which environment you would like to automate? Web - web applications in the browser
? With which browser should we start? Chrome
? Which framework do you want to use? Cucumber (https://cucumber.io/)
? Do you want to use a compiler? TypeScript (https://www.typescriptlang.org/)
? Do you want WebdriverIO to autogenerate some test files? Yes
? Where should these feature files be located?
/Users/shijithkannoth/Projects/POC/issue-reproduce/features/**/*.feature
? Where should these step definitions be located?
/Users/shijithkannoth/Projects/POC/issue-reproduce/features/step-definitions/steps.ts
? Do you want to use page objects (https://martinfowler.com/bliki/PageObject.html)? Yes
? Where are your page objects located?
/Users/shijithkannoth/Projects/POC/issue-reproduce/features/pageobjects/**/*.ts
? Which reporter do you want to use? spec, allure
? Do you want to add a plugin to your test setup?
? Do you want to add a service to your test setup?
? What is the base url? http://localhost
? Do you want me to run `npm install` Yes


Setting up TypeScript...
✔ Success!

Installing wdio packages:
- @wdio/local-runner@latest
- @wdio/cucumber-framework@latest
- @wdio/spec-reporter@latest
- @wdio/allure-reporter@latest
- ts-node
- typescript

added 136 packages, and audited 664 packages in 11s

93 packages are looking for funding
  run `npm fund` for details

2 critical severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
✔ Success!

Creating a WebdriverIO config file...
✔ Success!

Autogenerating test files...
✔ Success!

Adding "wdio" script to package.json.
✔ Success!

🤖 Successfully setup project at /Users/shijithkannoth/Projects/POC/issue-reproduce 🎉

Join our Discord Community Server and instantly find answers to your issues or queries. Or just join and say hi 👋!
  🔗 https://discord.webdriver.io

Visit the project on GitHub to report bugs 🐛 or raise feature requests 💡:
  🔗 https://github.com/webdriverio/webdriverio

To run your tests, execute:
$ cd /Users/shijithkannoth/Projects/POC/issue-reproduce
$ npm run wdio

$ npm install typescript ts-node --save-dev

npm install @wdio/cucumber-framework --save-dev

npx wdio wdio.conf.ts

**Comment by erwinheitzman** [comment]
> @jan-molak just fyi: this is the edge case that required me to set `-r ts-node/register` to the `NODE_FLAGS` environment. Do you happen to know any easy way to detect whether a project is running as ESM or CJS project? Because based on that we should set one or the other ts-node hook. There doesn't seem to be one for both.

I believe that you can check for the `type` property to be set to `module` in the `package.json` however it might be hard to find out the correct `package.json` file.

**Comment by erwinheitzman** [comment]
I think as a quick workaround we could introduce a --esm for running wdio with esm projects, when given we assume esm. Later we could deprecate and remove it when we feel like we have a stable setup to detect the difference. What do you think?

**Comment by christian-bromann** [comment]
> I just used the standard instruction given for typescript setup with cucumber.

@SKannoth this won't reproduce the issue. The tests are passing fine. Do you have a github repository we can close and test?

> we could introduce a --esm for running wdio with esm projects, when given we assume esm

@erwinheitzman I am not sure if people would understand when to use this flag or not.

**Comment by jan-molak** [comment]
> > @jan-molak just fyi: this is the edge case that required me to set `-r ts-node/register` to the `NODE_FLAGS` environment. Do you happen to know any easy way to detect whether a project is running as ESM or CJS project? Because based on that we should set one or the other ts-node hook. There doesn't seem to be one for both.
> 
> 
> 
> I believe that you can check for the `type` property to be set to `module` in the `package.json` however it might be hard to find out the correct `package.json` file.

Yeah, I suggested the same approach. I think me might need to rely on some simple heuristics:
- browser runner => ESM
- other runner and type: mudule in package.json => ESM
- anything else => CJS

The above list may or may not be complete. I agree with Christian, though, relying on a flag might be confusing, unless we wanted to have it act as an override of the loader detection mechanism 🤔

Regarding finding package.json, it feels like it might be something like:
- Look for package.json in current working directory Where wdio is launched
- if found, read the "type" field
- if file not found, either stop and assume CJS, or repeat the search in the parent directory (recursively, until you reach the root)

Alternatively, the search could be started in the parent directory of where wdio is installed (skipping any nested node modules sub-directories for the cases where wdio is installed as a dependency of another tool)

**Comment by jan-molak** [comment]
> @jan-molak what do you think using a package like https://github.com/unjs/mlly .. I am sure we are not the first people that try to solve this 😉 

Mlly seems to be doing more than we need, like parsing the source code etc. I used read-pkg in the past and it does the job pretty well - https://www.npmjs.com/package/read-pkg

**Comment by christian-bromann** [comment]
mlly uses acorn which is a tiny, fast JavaScript parser written in JavaScript (0 deps). Adding it as a dependency won't add much baggage. It is being used by Vite and I feel more confident relying on their implementation than spending the time coming up with a custom approach.

**Comment by jan-molak** [comment]
> mlly uses acorn which is a tiny, fast JavaScript parser written in JavaScript (0 deps). Adding it as a dependency won't add much baggage. It is being used by Vite and I feel more confident relying on their implementation than spending the time coming up with a custom approach.

I don't mind either way as long as it works ☺️ 
We'll need solid integration tests in this area.

**Comment by christian-bromann** [comment]
This is great insights @andrew-nguyen-mcs , thanks for sharing!

**Comment by olivier-martin-sf** [comment]
Adding my 2 cents, I am hitting a similar error in a CommonJS package where the WDIO tests are authored in TypeScript using `import`. `ts-node` is responsible for compiling the tests on the fly to CJS and it works perfectly well with node 16, 18 and 19 but stop working from node >= 20.0.0. 

We use a GitHub Action matrix so the exact same set of tests are being run in parallel in CI. Node.js 16 and 18 works great but all the tests are failing under node.js 20. 

@christian-bromann do you mind elaborating more on the root cause of the issue? I tried to set `NODE_OPTIONS=-r ts-node/register` in from of the npm script but it didn't fix the issue for me.

**Comment by christian-bromann** [comment]
@olivier-martin-sf thanks for the input. The problem here is that in order to import the `wdio.conf.ts` we need to [restart](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-cli/src/commands/run.ts#L180C26-L180C62) the Node.js process with `--loader ts-node/esm/transpile-only` flag set which is required by `ts-node`. Now it seems that this fails for CJS projects. I added a `-r ts-node/register` flag but having both flags seem to mix up stack traces so we reverted that.

**Comment by sabrina-tw** [comment]
Just wanted to add that I'm facing the same issue after upgrading Node from version 18.12.1 to 20.5.1

Has there been a working temporary solution to this? 🙏🏼 
The `"type": "module"` and `"module": "ESNext"` did not work for my project.

**Comment by christian-bromann** [comment]
@sabrina-tw can you provide a minimal reproducible example?

**Comment by sceiler** [comment]
@christian-bromann 👋 
By coincidence I an running into the same issue. Here is my repo where you can reproduce it. https://github.com/sceiler/guestbook-wdio/tree/31ef6c486996521752b6ce112dfab9586a404b4a

npm is 9.6.7
wdio is 8.16.7
node is v18.17.1

The command I a running is `npm run wdio-github`

**Comment by erwinheitzman** [comment]
This should be resolved by updating to the latest version

**Comment by erwinheitzman** [comment]
@sceiler I have ran your setup with Node 18 and the latest versions of the wdio packages without issues, closing the issue due to inactivity after my last message.

**Comment by gvenkatnz5658** [comment]
> So, I'm not entirely sure if this will work for other people, but this is just what we did to get around this. I was at node version 20.3.1 and switched to 19.6.1, which allowed the tests to be run without the import error. Not completely sure if it's relevant, but thought it might be helpful.

I'm in the similar situation. Was on 18.16.1 and upgraded to 18.19.0, got the import error. So we had to use 18.16.1.
would greatly appreciate if anyone have a solid answer.

**Comment by gvenkatnz5658** [comment]
Thank you for your response.
Really appreciate it - glad it's actually a known issue..
